lib/pokerscore/hand.rb

```diff
@@ -38,11 +38,7 @@ def invalid_hand?(cards)
   end
 
   def flush?
-    flush = true
-    cards.each do |card|
-      flush = false if card.suit != cards.first.suit
-    end
-    flush
+    cards.map { |card| card.suit }.uniq.length == 1
```

Comment:
We can be even sneakier here by using some `to_proc` magic in Ruby: https://stackoverflow.com/a/1961118